src/seedu/addressbook/commands/SortCommand.java

```diff
@@ -0,0 +1,39 @@
+package seedu.addressbook.commands;
+
+import seedu.addressbook.common.Messages;
+import seedu.addressbook.data.person.ReadOnlyPerson;
+
+import java.util.List;
+
+/**
+ * Sorts the list of persons in alphabetical order by first name.
+ */
+public class SortCommand extends Command {
+
+    public static final String COMMAND_WORD = "sort";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Sorts the address book "
+            + "in alphabetical order by first name.\n"
+            + "Example: " + COMMAND_WORD;
+
+    public static final String MESSAGE_SUCCESS = "Address book has been sorted!";
+
+    @Override
+    public CommandResult execute() {
+        if (addressBook.getAllPersons().size() > 1) {
+            if (addressBook.getAllPersons().isSorted()) {
+                return new CommandResult(Messages.MESSAGE_ALREADY_SORTED);
+            } else {
+                addressBook.sort();
+                List<ReadOnlyPerson> allPersons = addressBook.getAllPersons().immutableListView();
+                return new CommandResult(MESSAGE_SUCCESS
+                        + "\n"
+                        + getMessageForPersonListShownSummary(allPersons),
+                        allPersons);
+            }
+        } else {
```

Comment:
![image](https://user-images.githubusercontent.com/12574756/36298819-ff734066-1334-11e8-825e-72e9938fd9dd.png)

Avoid arrow head code.
You can make a new function which does the sorting